src/MOI_wrapper.jl

```diff
@@ -2488,3 +2488,62 @@ function MOI.get(
     end
     return get_intattrelement(model.inner, "IISQConstr", _info(model, index).row) > 0
 end
+
+###
+### Constraint attributes
+###
+
+# Gurobi constraint attributes as documented at https://www.gurobi.com/documentation/8.1/refman/linear_constraint_attribut.html
+const CONSTR_ATTR_TYPE =
+    Dict( "Lazy" => Integer, "DStart" => Real, "IISConstr" => Integer,
+          "SARHSLow" => Real, "SARHSUp" => Real, "FarkasDual" => Real,
+          "Sense" => Char, "RHS" => Real, "ConstrName" => String, "Pi" => Real,
+          "Slack" => Real, " CBasis" => Integer )
+const GETTER_FOR_TYPE = Dict( Integer => Gurobi.get_intattrelement,
+                              Real => Gurobi.get_dblattrelement,
+                              Char => Gurobi.get_charattrelement,
+                              String => Gurobi.get_strattrelement)
+const SETTER_FOR_TYPE = Dict( Integer => Gurobi.set_intattrelement!,
+                              Real => Gurobi.set_dblattrelement!,
+                              Char => Gurobi.set_charattrelement!,
+                              String => Gurobi.set_strattrelement! )
+
+struct ConstraintAttribute <: MOI.AbstractConstraintAttribute
+    name::String
+end
+
+"""
+MOI.set(model::Optimizer, attr::ConstraintAttribute,
+        ci::MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}, <:Any},
+        value::T) where T
+
+Set a constraint attribute.
+    
+Checks that the attribute exists and that value is correctly typed, but lets
+Gurobi handle cases where an attribute's value cannot be set.
+
+Caveat: might fail due to incorrect type for an attribute that cannot be set
+anyway.
+"""
+function MOI.set(
+    model::Optimizer, attr::ConstraintAttribute,
+    ci::MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}, <:Any},
+    value::T
+) where T
+    @assert attr.name ∈ keys(CONSTR_ATTR_TYPE) "Unknown constraint attribute for Gurobi backend: $(attr.name)."
```

Comment:
You should throw `MOI.UnsupportedAttribute`. This check should also be in `MOI.supports(::Optimizer, ::ConstraintAttribute)`

Reply:
Done in https://github.com/JuliaOpt/Gurobi.jl/pull/243/commits/61dda72bb7bbfe065fcf009fbc3a8aaa63547eee